clients/pkg/logentry/stages/eventlogmessage.go

```diff
@@ -0,0 +1,164 @@
+package stages
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	defaultSource                = "message"
+	ErrEmptyEvtLogMsgStageConfig = "empty event log message stage configuration"
+)
+
+type EventLogMessageConfig struct {
+	Source            *string `mapstructure:"source"`
+	DropInvalidLabels bool    `mapstructure:"drop_invalid_labels"`
+	OverwriteExisting bool    `mapstructure:"overwrite_existing"`
+}
+
+type eventLogMessageStage struct {
+	cfg    *EventLogMessageConfig
+	logger log.Logger
+}
+
+// Create a event log message stage, including validating any supplied configuration
+func newEventLogMessageStage(logger log.Logger, config interface{}) (Stage, error) {
+	cfg, err := parseEventLogMessageConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	// validate config (i.e., check that source is a valid log label)
+	err = validateEventLogMessageConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &eventLogMessageStage{
+		cfg:    cfg,
+		logger: log.With(logger, "component", "stage", "type", "event_log_message"),
+	}, nil
+}
+
+// Parse the event log message configuration, creating a default configuration struct otherwise
+func parseEventLogMessageConfig(config interface{}) (*EventLogMessageConfig, error) {
+	cfg := &EventLogMessageConfig{}
+	err := mapstructure.Decode(config, cfg)
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+// Ensure a event log message configuration object is valid, checking that any specified source
+// is a valid label name, and setting default values if a nil config object is provided
+func validateEventLogMessageConfig(c *EventLogMessageConfig) error {
+	if c == nil {
+		return errors.New(ErrEmptyEvtLogMsgStageConfig)
+	}
+	if c.Source != nil && !model.LabelName(*c.Source).IsValid() {
+		return fmt.Errorf(ErrInvalidLabelName, *c.Source)
+	}
+	return nil
+}
+
+func (m *eventLogMessageStage) Run(in chan Entry) chan Entry {
+	out := make(chan Entry)
+	key := defaultSource
+	if m.cfg.Source != nil {
+		key = *m.cfg.Source
+	}
+	go func() {
+		defer close(out)
+		for e := range in {
+			err := m.processEntry(e.Extracted, key)
+			if err != nil {
+				continue
```

Comment:
Could you report the error in the logs as a warning?

Reply:
There's currently only a single context where err won't be nil (when getString returns an err) - I've now generated a warning when that occurs - so I think this should be covered (seems a little overkill to warn about it twice).